src/home_audit.ts

```diff
@@ -0,0 +1,807 @@
+        case 'mini_split':
+            if(!heating_type || ['heat_pump', 'gchp'].includes(heating_type)){
+                heat_cool_valid = false
+            }
+            break
+        case 'split_dx':
+            if(!heating_type || ['heat_pump', 'gchp', 'mini_split'].includes(heating_type)){
+                heat_cool_valid = false
+            }
+            break
+        case 'dec':
+            if(!heating_type || ['gchp'].includes(heating_type)){
+                heat_cool_valid = false
+            }
+            break
+    }
+    if(!heat_cool_valid) {
+        addErrorMessage(`systems/hvac/${index}/heating/type`, `${heating_type} is not an appropriate heating type with cooling type ${cooling_type}`)
+    }
+}
+
+/**
+ * Check that the efficiency method is valid for the heating type
+ */
+function checkHeatingEfficiencyValid({type, fuel_primary, efficiency_method}:HeatingSystem, index:number): void
+{
+    if(efficiency_method &&
+        ([null, undefined, 'baseboard', 'wood_stove', 'none'].includes(type) ||
+        (type === 'central_furnace' && fuel_primary === 'electric'))
+    ) {
+        addErrorMessage(`systems/hvac/${index}/heating/efficiency_method`, `Efficiency method should not be set if heating type is "central furnace" and fuel is "electric", or if heating type is "baseboard", "wood stove", "none", or empty`)
+    }
+    if(efficiency_method === 'shipment_weighted') {
+        if(type === 'wall_furnace' && fuel_primary !== 'natural_gas') {
+            addErrorMessage(`systems/hvac/${index}/heating/efficiency_method`, `Efficiency method must be "user" if heating type is "wall_furnace" and fuel is not "natural_gas"`)
+        }
+        if(type && ['mini_split', 'gchp'].includes(type)) {
+            addErrorMessage(`systems/hvac/${index}/heating/efficiency_method`, `Heating efficiency method must be "user" when heating type is "${type}"`)
+        }
+    }
+}
+
+/**
+ * Check that the efficiency method is valid for the cooling type
+ */
+function checkCoolingEfficiencyValid(cooling:CoolingSystem, index:number): void
+{
+    const {type, efficiency_method} = cooling
+    if(efficiency_method && [null, undefined, 'none', 'dec'].includes(type)) {
+        addErrorMessage(`systems/hvac/${index}/cooling/efficiency_method`, `Efficiency method should not be set if cooling type is "none", "direct evaporative cooler", or empty`)
+    }
+    if(efficiency_method !== 'user' && type && ['mini_split', 'gchp'].includes(type)) {
+        addErrorMessage(`systems/hvac/${index}/cooling/efficiency_method`, `Cooling efficiency must be 'user' when type is '${type}'`)
+    }
+}
+
+/**
+ * Check that the HVAC system is of a valid year
+ */
+const MIN_SYSTEM_YEAR = 1970
+const MAX_SYSTEM_YEAR = (new Date()).getFullYear() // this year
+function checkSystemYear({year}:HeatingSystem|CoolingSystem, name:'heating'|'cooling', index:number): void
+{
+    if(year === undefined || MIN_SYSTEM_YEAR > year || year > MAX_SYSTEM_YEAR) {
```

Comment:
This check is incorrect; `year` is nullable in the schema (in favor of user entered efficiency).  If year isn't provided though, this error will always return.

Reply:
Changed this to `if(year && (MIN_SYSTEM_YEAR > year || year > MAX_SYSTEM_YEAR))`